MonoGame.Framework/BoundingBox.cs

```diff
@@ -132,20 +130,20 @@ public ContainmentType Contains(BoundingFrustum frustum)
                 return ContainmentType.Intersects;
 
 
-            // If we get here, it means the first (and only) point we checked was actually contained in the frustum.
-            // So we assume that all other points will also be contained. If one of the points is disjoint, we can
-            // exit immediately saying that the result is Intersects
+            // If we get here it means the first (and only) point we checked was outside the frustum, so we assume
+            // that all other points will be as well.
+            // If one of the points is contained, we can exit immediately saying that the result is Intersects
             i++;
             for (; i < corners.Length; i++)
             {
                 this.Contains(ref corners[i], out contained);
-                if (contained != ContainmentType.Contains)
+                if (contained == ContainmentType.Contains)
                     return ContainmentType.Intersects;
 
             }
 
             // If we get here, then we know all the points were actually contained, therefore result is Contains

```

Comment:
This comment should also be updated to something like: If we get here, then we know all the points are actually outside the box, therefore result is Disjoint.

On that note, is it not possible to have an intersection between a BoundingBox and BoundingFrustum where none of their corners are contained within the other volume? The Frustum is not box shaped and is not axis aligned, so a corner could technically pass through the whole of the other volume and not be contained within while still intersecting.

Reply:
I had the same thought, but if the Frustum's bigger than the Box then it can't really be contained by it. Maybe that case should return an Intersect instead though

Reply:
You could have an intersection where the Fustrum's bigger than the box, but if you're calling `.Contains()` on the box while it's entirely contained in the Fustrum it makes sense to still return Intersects I think.
So, I think if all the Fustrum's points are outside the box, we can check to see if the box is contained in the Fustrum, and if we get anything but Disjoint return Intersects.